src/store/swapOrderSlice.ts

```diff
@@ -80,11 +82,18 @@ export const swapOrderStatusListener = (listenerMiddleware: typeof listenerMiddl
       if (oldStatus === newStatus || newStatus === undefined) {
         return
       }
+      const safeInfo = selectSafeInfo(listenerApi.getState())
+
+      let link = null
+      if (swapOrder.txId && safeInfo.data?.chainId && safeInfo.data?.address) {
+        const chainInfo = selectChainById(listenerApi.getState(), safeInfo.data?.chainId)
+        if (chainInfo !== undefined) {
+          link = getTxLink(swapOrder.txId, chainInfo, safeInfo.data?.address.value)
```

Comment:
I feel like it would be more re-usable to add a new optional field `txId` to the `notificationSlice`.
Then our redux middleware does not need to know how to create URIs and fetch all this data here. The Notification component could handle that.

Reply:
yes, I agree! I was thinking about the same. I think that we also need to pass the safe address this way we can fix the bug where we display a notification for the wrong safe.

Doing this kind of modifications in this PR however seemed like to much of a refactoring and wouldn't be clear why it was done. 

Also there are some instances where we have more complex links than just to the transaction:
![grafik](https://github.com/safe-global/safe-wallet-web/assets/693770/27a3c7ba-3dc2-4feb-8302-a04c363da6eb)

So, you are suggesting we keep a txId and link. If no link is passed, but a txId exist, we try to link to the transaction?

Reply:
Hmm we can also for now leave it as is.
I think reworking the whole notifications and merging them with pendingTxs is also still an idea floating around and would be the proper way of doing this.

I'll approve for now.